backend/src/db/migrations/20230920200620_record_achieved_achievements_in_users_achievements_table.ts

```diff
@@ -0,0 +1,38 @@
+import { type Knex } from 'knex';
+
+const TableName = {
+  USERS_ACHIEVEMENTS: 'users_achievements',
+  ACHIEVEMENTS: 'achievements',
+  ARTICLES: 'articles',
+  COMMENTS: 'comments',
+} as const;
+
+const up = (knex: Knex): Promise<void> => {
+  return knex.raw(`
+    WITH UsersCounts AS (
+      SELECT t.user_id, t.reference_table, COUNT(*)
+      FROM (
+        SELECT c.user_id, c.id, a.reference_table
+        FROM ${TableName.COMMENTS} c
+        JOIN ${TableName.ACHIEVEMENTS} a ON a.reference_table = '${TableName.COMMENTS}'
+        GROUP BY c.user_id, c.id, a.reference_table
+        UNION ALL
+        SELECT ar.user_id, ar.id, a.reference_table
+        FROM ${TableName.ARTICLES} ar
+        JOIN ${TableName.ACHIEVEMENTS} a ON a.reference_table = '${TableName.ARTICLES}'
+        GROUP BY ar.user_id, ar.id, a.reference_table
+      ) t
```

Comment:
Let's get rid of shortens, as with them it's hard to understand what table is actually being referenced 
```suggestion
      ) grouped_reference_tables_counts
```

Reply:
Get rid of all the shortens?